### PR TITLE
fix: MQTT code coverage.

### PR DESCRIPTION
**Pull request by abhidixi11**
Improve MQTT code coverage - for iot_mqtt_api.c

*Issue #, if available:*

*Description of changes:*
Increase code coverage for iot_mqtt_api.c.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: libraries/standard/mqtt/test/access/iot_test_access_mqtt_api.c [hunk 1]
```diff
@@ -33,6 +33,13 @@ _mqttConnection_t * IotTestMqtt_createMqttConnection( bool awsIotMqttMode,
                                                       const IotMqttNetworkInfo_t * pNetworkInfo,
                                                       uint16_t keepAliveSeconds );
 
+void IotTestMqtt_mqttOperation_tryDestroy( void * pData );
+
+IotNetworkError_t IotTestMqtt_createNetworkConnection( const IotMqttNetworkInfo_t * pNetworkInfo,
+                                                       IotNetworkConnection_t * pNetworkConnection,
+                                                       bool * pCreatedNewNetworkConnection );
+
+
```
Comment (leegeth) [inline libraries/standard/mqtt/test/access/iot_test_access_mqtt_api.c]: why are these functions forward declared? Include of iot_test_access_mqtt.h will do that.

File: libraries/standard/mqtt/test/unit/iot_tests_mqtt_api.c [hunk 2]
```diff
@@ -1073,6 +1123,33 @@ TEST( MQTT_Unit_API, DisconnectMallocFail )
...
+    networkInfo.createNetworkConnection = false;
+    networkInfo.disconnectCallback.pCallbackContext = &expectedReason;
+    networkInfo.disconnectCallback.function = _disconnectCallback;
+
+    /* Set up a mocked MQTT connection. */
+    TEST_ASSERT_EQUAL_INT( true, IotTest_MqttMockInit( &mqttConnection ) );
+
+    mqttConnection->disconnected = true;
+    IotMqtt_Disconnect( mqttConnection, 0 );
+    /* There should not be any asserts or test memory leak errors */
```
Comment (leegeth) [inline libraries/standard/mqtt/test/unit/iot_tests_mqtt_api.c]: What is this test doing? How will it validate if there is any failure or not?

File: libraries/standard/mqtt/test/access/iot_test_access_mqtt_api.c (hunk 1 above)
Comment (abhidixi11) [reply libraries/standard/mqtt/test/access/iot_test_access_mqtt_api.c]: I agree. It was already there and I just kept the same format.

File: libraries/standard/mqtt/src/iot_mqtt_api.c [hunk 3]
```diff
@@ -1682,6 +1682,7 @@ IotMqttError_t IotMqtt_Wait( IotMqttOperation_t operation,
     if( _IotMqtt_ValidateOperation( operation ) == false )
     {
         status = IOT_MQTT_BAD_PARAMETER;
+        goto cleanup;
```
Comment (gkwicker) [inline libraries/standard/mqtt/src/iot_mqtt_api.c]: The commit message isn't accurate since we are changing functionality here.  Prior to this change `pMqttConnection` would have been updated on line 1689.  I'd either split this into two separate commits, or change the description to say "Fixed error handling case in IotMqtt_Wait and increased code coverage".
